homeassistant/components/switch/switchbot.py

```diff
@@ -0,0 +1,93 @@
+"""
+Support for Switchbot.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.switchbot
+"""
+import binascii
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME, CONF_MAC
+
+REQUIREMENTS = ['bluepy==1.1.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Switchbot'
+UUID = "cba20d00-224d-11e6-9fb8-0002a5d5c51b"
+HANDLE = "cba20002-224d-11e6-9fb8-0002a5d5c51b"
+ON_KEY = "570101"
+OFF_KEY = "570102"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MAC): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Perform the setup for Switchbot devices."""
+    name = config.get(CONF_NAME)
+    mac_addr = config.get(CONF_MAC)
+    add_devices([SwitchBot(mac_addr, name)])
+
+
+class SwitchBot(SwitchDevice):
+    """Representation of a Switchbot."""
+
+    def __init__(self, mac, name) -> None:
+        """Initialize the Switchbot."""
+        self._state = False
+        self._name = name
+        self._mac = mac
+        self._device = None
+
+    def _sendpacket(self, key, retry=2) -> bool:
```

Comment:
This belongs in a standalone library.

Reply:
Ok, I thought it might be ok since it so few lines.
But I will make a standalone library.